addons/product/models/product_pricelist.py

```diff
@@ -592,18 +594,25 @@ def _compute_price(self, price, price_uom, product, quantity=1.0, partner=False)
             # complete formula
             price_limit = price
             price = (price - (price * (self.price_discount / 100))) or 0.0
+            product_cur = product.cost_currency_id if self.base == 'standard_price' else product.currency_id
```

Comment:
```suggestion
            if self.base == 'standard_price':
                price_currency = product.cost_currency_id
            elif self.base == 'pricelist':
                price_currency = self.currency_id  # Already converted before to the pricelist currency
            else:
                price_currency = product.currency_id
```
My bad, forgot to mention the case where a rule is based on the result of another pricelist.